binary variable length data (LVAR) support

PieterInepro opened the pull request:
The current variable length field implementation only supports ascii strings, LVAR interpretation is incorrect and incomplete.

Fix the LVAR interpretation, and add support for ascii and binary data types.

PieterInepro committed:
mbus-protocol: fix variable length data field parsing

-use LVAR byte interpretation according to EN13757-3:2018 Table 5
-keep the LVAR byte as first record data byte, as it defines the data
type (ascii / BCD / binary)
-print variable length binary data
(also print variable length BCD data as binary for now)

PieterInepro committed:
mbus-protocol: print binary data LSB-first

EN13757-3 requires all multi-byte fields to be transmitted Least
Significant Byte first.

NOTE: mbus_data_bin_decode is used for variable length data (which uses
LSB-first) as well as manufacturer specific data.
Assume manufacturer specific data will also use LSB order in compliance
with EN13757-3. However, this is up to the manufacturer definition of
course.

lategoodbye commented:
Thanks for working on this. So there is currently no coverage by the existing M-Bus test frames?

PieterInepro commented:
Now that you mention it, I was not aware of the test frames.
Just ran a quick test on all test-frames hex files, there are a few with variable length data fields, but all of them are text strings.

Should I add a test frame with binary variable length data for reference?

lategoodbye commented:
Yes, please

PieterInepro committed:
mbus-protocol-aux: add binary variable length data field support

mbus_variable_value_decode() is used for 'normalized' xml output
generation

NOTE: BCD variable length data fields are also printed as binary for now

PieterInepro committed:
test-frames: add 16 byte binary LVAR example

PieterInepro commented:
I have added a 16-byte binary LVAR example.
Doing so, I noticed that the normalized xml output was using a different formatting function, which also needed to be adjusted to the new format.

lategoodbye commented on test/test-frames/example_binary16_lvar.xml:
```diff
@@ -0,0 +1,23 @@
+LVAR f0
```
Is this some kind of debug?
Unfortunately this makes the XML invalid.

PieterInepro replied:
yes you are right, sorry I missed that.

PieterInepro committed:
example_binary16_lvar.xml: remove debug log output

a debug log output line had accidentally been prepended to the generated
xml
